src/cdk-experimental/menu/menu.ts

```diff
@@ -34,4 +43,26 @@ export class CdkMenu extends CdkMenuGroup {
 
   /** Event emitted when the menu is closed. */
   @Output() readonly closed: EventEmitter<void | 'click' | 'tab' | 'escape'> = new EventEmitter();
+
+  /** List of nested CdkMenuGroup elements */
+  @ContentChildren(CdkMenuGroup, {descendants: true})
+  private readonly _nestedGroups: QueryList<CdkMenuGroup>;
+
+  ngAfterContentInit() {
+    super.ngAfterContentInit();
+
+    // If there are nested MenuGroup elements within this group
+    // we complete the change emitter in order to ensure that
+    // the this menu does not emit change events.
+    if (this._hasNestedGroups()) {
```

Comment:
Is the if-then necessary here? I would expect the behavior to be the same even without it, assuming _nestedGroups.changes emits immediately if there are nested groups from the beginning. (If that assumption isn't correct, then it does become necessary. I might add a comment explaining this, if that's the case.)

Reply:
Here I'm saying, if we have groups from the start lets close the emitter. Otherwise we don't have any on initial mount, but if some do show up, lets disable the emitter once they do.

Reply:
Right, so could this work without the if statement, by simply relying on the subscription?

Reply:
There's unfortunately no way to subscribe to the QueryList in such a way where your callback will get executed when the initial children are mounted. At the time you call subscribe the QueryList will have elements if they are already visible in the Component. Hence any logic needs to perform some logic on the existing children (if any) and subscribe to any future changes. 

For example, if we have a single CdkMenuGroup with no conditional mounting the QueryList will contain that one component  when `ngAfterContentInit` executes but the changes emitter won't emit until another is added, or the current one is removed. That means that if there is only ever one CdkMenuGroup, the changes emitter won't execute at all.